service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java

```diff
@@ -167,4 +171,11 @@ private boolean filterOutListResourceType(Reference reference) {
     private boolean checkIfReferenceIsObservation(Reference reference) {
         return reference.getReferenceElement().getResourceType().equals(ResourceType.Observation.name());
     }
+
+    private Optional<String> buildCode(Condition condition) {
+        if (condition.hasCode()) {
+            return Optional.of(codeableConceptCdMapper.mapCodeableConceptToCd(condition.getCode()));
```

Comment:
I don't think mapCodeableConceptToCd should ever return null. Do we need to use Optional here?